src/GrantRegistrar.sol

```diff
@@ -0,0 +1,205 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.19;
+
+import { Ownable } from 'openzeppelin-contracts/contracts/access/Ownable.sol';
+import { IWorldIDGroups } from 'world-id-contracts/interfaces/IWorldIDGroups.sol';
+import { Ownable2Step } from 'openzeppelin-contracts/contracts/access/Ownable2Step.sol';
+
+/// @title GrantRegistrar
+/// @author Worldcoin
+contract GrantRegistrar is Ownable2Step {
+  ///////////////////////////////////////////////////////////////////////////////
+  ///                              CONFIG STORAGE                            ///
+  //////////////////////////////////////////////////////////////////////////////
+
+  /// @dev The WorldID router instance that will be used for managing groups and verifying proofs
+  IWorldIDGroups public worldIdRouter;
+
+  /// @dev The World ID group whose participants can verify with this contract.
+  uint256 public groupId;
+
+  /// @dev The World ID nullifier hash that will be used to verify proofs.
+  uint256 immutable externalNullifierHash;
+
+  /// @dev The maximum grant ID that can be claimed without re-verifying.
+  uint256 public maxGrantId;
+
+  /// @dev The current grant ID.
+  uint256 public currentGrantId;
```

Comment:
We should get this from the grant contract: `grant.getCurrentId()`

Reply:
The main issue with this is that the `RecurringGrantDrop` contract depends on this one (and that one depends on `IGrant`, making it depend on this one implicitly as well). What would you recommend to do here?